crates/subspace-erasure-coding/src/tests.rs

```diff
@@ -0,0 +1,109 @@
+use crate::ErasureCoding;
+use std::iter;
+use std::num::NonZeroUsize;
+use subspace_core_primitives::Scalar;
+
+// TODO: This could have been done in-place, once implemented can be exposed as a utility
+fn concatenated_to_interleaved<T>(input: Vec<T>) -> Vec<T>
+where
+    T: Clone,
+{
+    if input.len() <= 1 {
+        return input;
+    }
+
+    let (first_half, second_half) = input.split_at(input.len() / 2);
+
+    first_half
```

Comment:
We lose the last element when `input.len()` is even. If it's intentional - please, add a comment before exposing it as a utility.

Reply:
Number of elements is always a power of 2. The case with 0 and 1 is covered at the beginning of the function.

Reply:
It's OK for tests, but if you're going to expose it as you mentioned in TODO, please add a comment for that, because this is unexpected behaviour.

Reply:
There is a TODO about it, it'll be done as part of that TODO if ever.